localstack-core/localstack/services/events/target.py

```diff
@@ -75,23 +77,28 @@ def get_template_replacements(
 
 
 def replace_template_placeholders(
-    template: str, replacements: dict[str, Any], is_json: bool
+    template: str, replacements: dict[str, Any], is_json_template: bool
 ) -> TransformedEvent:
     """Replace placeholders defined by <key> in the template with the values from the replacements dict.
     Can handle single template string or template dict."""
 
     def replace_placeholder(match):
         key = match.group(1)
         value = replacements.get(key, match.group(0))  # handle non defined placeholders
-        if is_json:
-            return to_json_str(value)
+        if is_json_template:
+            if is_nested_in_string(template, match):
+                return value  # Return raw string for nested placeholders
+            else:
+                return to_json_str(value)
         if isinstance(value, datetime.datetime):
             return event_time_to_time_string(value)
+        if isinstance(value, dict):
+            return dict_to_simple_string(value)
```

Comment:
If I understand you correct this scenario should be covered by this test cast `'{"method": "PUT", "path": "users-service/users/<payload>", "bod": "<payload>"},` since `payload` is a dictionary and it is set in the middle part in the string `"users-service/users/<payload>"`

Reply:
This will interestingly fail for AWS since it will not be correctly transformed since it would result in a string that is not a json string but the system expects a json string

I added a test for this failure also